doc/California-SB237/California-SB237.md

```diff
@@ -0,0 +1,238 @@
+Not relevant
+
+
+###  1.11. <a name='WarmbootandFastbootDesignImpact'></a>Warmboot and Fastboot Design Impact
+The feature can be triggered after sonic upgrade and warm reboot and feature doesn't affect trafic.
+
+###  1.12. <a name='RestrictionsLimitations'></a>Restrictions/Limitations
+The California law feature is not supported on remote AAA.
+
+LDAP/Radius/Tacacs is under customer responsibility.
+
+###  1.13. <a name='UpgradeFlow'></a>Upgrade Flow
+After install new image with sonic-install 1st boot flow is triggered and users will be forced to change their passwords
+
+
+###  1.14. <a name='TestPlan'></a>Test Plan
+####  1.14.1. <a name='UnitTestcases'></a>Unit Test cases
+Unit tests do not cover login and 1st boot flow
+
+
+####  1.14.2. <a name='SystemTestcases'></a>System Test cases
+Force 1st boot in next ways:
+- Onie install
+- Image update reboot/warm boot
+Check affecting password hardening feature:
+- Do 1st boot and test password change force with password hardening disabled
+- Do 1st boot and test password change force with password hardening enabled
+  Check password hardening age is not affected
+
+
+ #### 1.14.3. <a name='PasswordChangeFlow'></a>Pasword Change Flow
+
+ Example of password change during 1st login.
+
+    ssh admin@sonic-switch
+	The authenticity of host 'sonic-switch (111.111.111.111)' can't be established.
+	RSA key fingerprint is SHA256:l7jfE2zxvJ2PQNJZNB3ErqA/LvTUFi1tTlhDZTViPDI.
+	RSA key fingerprint is MD5:08:8c:5a:8a:7a:a7:65:13:5a:71:66:4c:30:2b:5c:ba.
+	Are you sure you want to continue connecting (yes/no)? yes
+	Warning: Permanently added 'sonic-switch,10.210.26.51' (RSA) to the list of known hosts.
+	admin@sonic-switch's password:
+	You are required to change your password immediately (administrator enforced).
+	You are required to change your password immediately (administrator enforced).
+	Linux sonic-switch 5.10.0-12-2-amd64 #1 SMP Debian 5.10.103-1 (2022-03-07) x86_64
+	You are on
+	____   ___  _   _ _  ____
+	/ ___| / _ \| \ | (_)/ ___|
+	\___ \| | | |  \| | | |
+	___) | |_| | |\  | | |___
+	|____/ \___/|_| \_|_|\____|
+
+	-- Software for Open Networking in the Cloud --
+
+	Unauthorized access and/or use are prohibited.
+	All access and/or use are subject to monitoring.
+
+	Help:    http://azure.github.io/SONiC/
+
+	Last login: Sat Aug 13 13:31:19 2022 from 10.209.102.100
+	WARNING: Your password has expired.
+	You must change your password now and login again!
+	Changing password for admin.
+	Current password:
+	New password:
+	Retype new password:
+	The password has not been changed.
+	New password:
+	Retype new password:
+	passwd: password updated successfully
+	Connection to sonic-switch closed.
+
+
+###  1.15. <a name='rdPartyComponents'></a>3rd Party Components
+In this section, you can find options used for password expiring with chage and pwhistory
+
+####  1.15.1. <a name='PWForceExpire'></a>PW Force Expiration
+
+Tool for password expiration:
```

Comment:
>expiration [](http://example.com/codeflow?start=18&length=10)

I am considering more use cases which requiring CLI:
1. Force all user password expiring
2. Force all user password not expiring any more

In real life, the manufactory and customers are login the device for different purpose:
1. the image is build with CHANGE_DEFAULT_PASSWORD turn on.
2. the manufactory test team will login and change password, and continue test
3. (missing) At the end of test, the manufactory test team will force all user password expiring
4. customers got the device, and login, and change password
5. customer start use it in normal way.